### PR TITLE
feat: add smali

### PR DESCRIPTION
**Pull request by amaanq**
This PR will add smali support, which I've forked from https://git.sr.ht/~yotam/tree-sitter-smali to https://github.com/amaanq/tree-sitter-smali, and have updated to fix a couple bugs. 

Thanks @ObserverOfTime for the queries, they seem good to me and I'm still all-too-new to this, so if whenever I notice something feels a bit off, I'll edit the query appropriately.

Paging @theHamsta for a review as he was the one that gave me the courage to maintain smali myself and submit a new PR from #3839 :)

Also, obvious note but it is of course experimental :laughing: 
-Amaan

### CODE REVIEW

**Comment by ObserverOfTime** [comment]
Can you merge https://git.sr.ht/~chronobserver/tree-sitter-smali/commit/b8b9adff29f95daf15696fa9fdacf190f415ef00 into your fork? I can't submit a PR at this time.

**Comment by amaanq** [comment]
Done, see: https://github.com/amaanq/tree-sitter-smali/commit/d7f535e176c928d33b0e202dd808ac247cacf2ff

**Comment by ObserverOfTime** [comment]
You should uncomment the character query too.

**Comment by theHamsta** [comment]
@amaanq I think you need to enable PRs in your forks settings. They might be disabled for forks.

**Comment by ObserverOfTime** [comment]
> I think you need to enable PRs in your forks settings. They might be disabled for forks.

No, I just didn't have my laptop at hand.

File: queries/smali/highlights.scm
Comment (ObserverOfTime) [inline queries/smali/highlights.scm:97]:
```suggestion
; unsupported
(character_literal) @character
```

**Comment by amaanq** [comment]
Done...force pushed since I had types `feaT` not `feat`..anything else @ObserverOfTime?

File: queries/smali/highlights.scm
Comment (ObserverOfTime) [inline queries/smali/highlights.scm:96]: It's not unsupported anymore :)

File: queries/smali/highlights.scm
Comment (theHamsta) [inline queries/smali/highlights.scm:82]: I don't know Smali. Could you check whether `@keyword.function` apply to any of these (.e.g `end_method`)

**Review by theHamsta — APPROVED** [review]

File: queries/smali/highlights.scm
Comment (ObserverOfTime) [reply queries/smali/highlights.scm:82]: It could be used for `.method` and `end_method` but I think it'd be better to keep them consistent with the rest.

Comment (amaanq) [reply queries/smali/highlights.scm:82]: .method and .end_method make sense to me to have as @keyword.function, even if syntactically it's the same as the others

Comment (ObserverOfTime) [reply queries/smali/highlights.scm:82]: Up to you.

**Comment by theHamsta** [comment]
@amaanq @ObserverOfTime is this ready to merge?

**Comment by ObserverOfTime** [comment]
Sure.

**Comment by amaanq** [comment]
give me ~1 hour and I'll update the .method and end_method directives.

**Comment by amaanq** [comment]
Ok, I'm ready to merge pending both @ObserverOfTime's and @theHamsta's approval

**Comment by theHamsta** [comment]
Thank you maintaining smali!

**Comment by amaanq** [comment]
No problem, I've been wanting it for a long time now but never knew where to start 😆

**Review by ObserverOfTime — APPROVED** [review]

**Comment by amaanq** [comment]
I could be totally missing something...but I have no highlights after installing smali like I'd normally expect in any other language

Smali:
image

Any other language:
image

**Comment by ObserverOfTime** [comment]
smali is not recognised by Vim. You can use this for now:
```lua
vim.filetype.add {
  extension = {
    smali = 'smali'
  }
}
```

**Comment by amaanq** [comment]
oh, great that fixed it! any idea where that feature is added for most files and/or where I should submit a PR to add smali? neovim source, vim source..?

**Pull request merged**
